wp/lib/bap_wp/src/compare.ml

```diff
@@ -35,13 +38,41 @@ let set_to_eqs (env1 : Env.t) (env2 : Env.t) (vars : Var.Set.t) : Constr.t list
     ~f:(fun (eqs, env1, env2) v ->
         let var1, env1 = Env.get_var env1 v in
         let var2, env2 = Env.get_var env2 v in
-        let eq = Bool.mk_eq ctx var1 var2
-                 |> Constr.mk_goal
-                   (Format.sprintf "%s = %s" (Expr.to_string var1) (Expr.to_string var2))
-                 |> Constr.mk_constr
-        in (eq::eqs, env1, env2)
+        (* We do not compare memory when we assume that memory is equal at an offset. *)
```

Comment:
Should we *ever* compare memory by default?

Reply:
This is where I was unsure,  but I think it is reasonable to assume that memory is not going to be equal by default. We still get the same results whether we add the memory comparison to the hypothesis or not.